c10/core/TensorTypeSet.h

```diff
@@ -33,7 +33,19 @@ namespace c10 {
 // An undefined tensor is one with an empty tensor type set.
 class TensorTypeSet {
 public:
-  TensorTypeSet() {}
+  enum Full { FULL };
+  enum Raw { RAW };
+
+  // NB: default constructor representation as zero is MANDATORY as
+  // use of TensorTypeSet in TLS requires this.
+  TensorTypeSet()
+    : repr_(0) {}
+  TensorTypeSet(Full)
+    : repr_(-1) {}
+  // Public version of TensorTypeSet(uint64_t) API; external users
+  // must be explicit when they do this!
+  TensorTypeSet(Raw, uint64_t x)
```

Comment:
interesting way to distinguish this, why not `static from_raw` or something?

Reply:
actually, this seems more discoverable, I like it

Reply:
Also, sometimes you can use a constructor where you cannot use a static method.